### PR TITLE
[BUGFIX] Fix outsider reputation displaying incorrectly in leader's den

### LINKED ISSUE #3158: [BUG] Outsiders Reputation Displays Incorrectly in Leader's Den
### Type of bug

Code - general bugs related to how the game runs and responds to you.

### Description

My Clan has a hostile relationship with outsiders, proved by the result that is locked behind hostile relationships. However, the leader's den Outsiders screen still displays "Outsiders view your Clan as welcoming".

### Grade

Unimportant - This bug has little to no impact on play.

### Steps to reproduce the behavior

1. Gain a hostile relationship with outsiders.
2. Check Leader's Den Outsiders screen.

### Patrol ID + Outcome Print (if applicable)

_No response_

### Commit # or Game Version Number

0.12.2

### Additional context

It's possible this arose from the "Threatening Outsiders" function in the warrior's den, but my cats also antagonize outsiders on patrols.

### Screenshots

image
welcoming reputation

image
hostile reputation

These are from the same moon.

### PR DESCRIPTION
**Pull request by rustykitty**
## About The Pull Request
Added a few checks that prevent the outsider reputation from going below 0. 

## Why This Is Good For ClanGen
N/A

## Linked Issues
Resolves #3158

## Proof of Testing
Screenshot 2024-12-19 at 9 27 40 PM
Screenshot 2024-12-19 at 9 27 54 PM
Screenshot 2024-12-19 at 9 28 04 PM

## Changelog/Credits
Outsider reputation no longer displays as "welcoming" when too low.

### CODE REVIEW

**Comment by j-gynn** [comment]
Targeting the wrong branch - this has files from `development` in it.

**Review by j-gynn — DISMISSED** [review]
Built from `development` but targeting `release-0.12.0`

File: scripts/clan.py [hunk 1]
```diff
@@ -813,7 +813,7 @@ def load_clan_json(self):
         )
         game.clan.post_initialization_functions()
 
-        game.clan.reputation = int(clan_data["reputation"])
+        game.clan.reputation = min(0, int(clan_data["reputation"]))
```
Comment (filvas2) [inline scripts/clan.py]: As clan reputation is 0-100, you could use 
`game.clan.reputation = clamp(int(clan_data["reputation"]),0,100)`

File: scripts/utility.py [hunk 2]
```diff
@@ -278,6 +278,8 @@ def change_clan_reputation(difference):
     will change the Clan's reputation with outsider cats according to the difference parameter.
     """
     game.clan.reputation += difference
```
Comment (filvas2) [inline scripts/utility.py:280]: This is also an option
`game.clan.reputation = clamp(game.clan.reputation + difference, 0 , 100) `

File: scripts/clan.py (hunk 1 above)
Comment (rustykitty) [reply scripts/clan.py]: I don't see a clamp function being used anywhere else in the code and the code I put here is based on the `change_clan_relations` function below.

Comment (filvas2) [reply scripts/clan.py]: Ah, my bad, it's only in development branch
image

**Review by scribblecrumb — APPROVED** [review]

**Pull request merged**